File: java/src/main/java/kata/BallSpinAdapter.java
package kata;

public class BallSpinAdapter implements BallSpinner {

  private static final int DEFAULT_TIME = 20000;
  private final int timeInMilliseconds;

  public BallSpinAdapter(int timeInMilliseconds) {
    this.timeInMilliseconds = timeInMilliseconds;
  }

  public BallSpinAdapter() {
    this(DEFAULT_TIME);
  }

  @Override
  public void spin() {
    try {
      Thread.sleep(timeInMilliseconds);
    } catch (InterruptedException e) {
    }
  }

}

File: python/src/kata/ball_spin_adapter.py
import time
from .ball_spinner import BallSpinner

class BallSpinAdapter(BallSpinner):
    DEFAULT_TIME = 20000

    def __init__(self, time_in_milliseconds: int = DEFAULT_TIME):
        self.time_in_milliseconds = time_in_milliseconds

    def spin(self) -> None:
        time.sleep(self.time_in_milliseconds / 1000)  # Convert milliseconds to seconds 